content/blog/postprocessing-preview/index.Rmd

```diff
@@ -0,0 +1,236 @@
+
+```{r}
+#| label: install-packages
+#| eval: false
+pak::pak(
+  paste0(
+    "tidymodels/",
+    c("tune", "workflows", "rsample", "tailor")
+  )
+)
+```
+
+Now, we load packages with those developmental versions installed.
+
+```{r}
+#| label: load-packages
+#| message: false
+library(tidymodels)
+library(probably)
+library(tailor)
+```
+
+Existing tidymodels users might have spotted something funky already; who is this tailor character?
+
+## Meet tailor👋
+
+The tailor package introduces tailor objects, which compose iterative adjustments to model predictions. tailor is to postprocessing as recipes is to preprocessing; applying your mental model of recipes to tailor should get you a good bit of the way there.
+
+| Tool | Applied to\... | Initialize with\... | Composes\... | Train with\... | Predict with\... |
+|------------|------------|------------|------------|------------|------------|
+| recipes | Training data | `recipe()` | `step_*()`s | `prep()` | `bake()` |
+| tailor | Model predictions | `tailor()` | `adjust_*()`ments | `fit()` | `predict()` |
+
+First, users can initialize a tailor object with `tailor()`.
+
+```{r}
+#| label: tailor
+tailor()
+```
+
+Tailors compose "adjustments," analogous to steps from the recipes package.
+
+```{r}
+#| label: tailor-adjust-1
+tailor() %>%
+  adjust_probability_threshold(threshold = .7)
+```
+
+As an example, we'll apply this tailor to the `two_class_example` data made available after loading tidymodels.
+
+```{r}
+#| label: two-class-example
+head(two_class_example)
+```
+
+This data gives the true value of an outcome variable `truth` as well as predicted probabilities (`Class1` and `Class2`). The hard class predictions, in predicted, are `"Class1"` if the probability assigned to `"Class1"` is above .5, and `"Class2"` otherwise.
+
+The model predicts `"Class1"` more often than it does `"Class2"`.
+
+```{r}
+#| label: count-two-class-example
+two_class_example %>% count(predicted)
+```
+
+If we wanted the model to predict `"Class2"` more often, we could increase the probability threshold assigned to `"Class1"` above which the hard class prediction will be `"Class1"`. In the tailor package, this adjustment is implemented in `adjust_probability_threshold()`, which can be situated in a tailor object.
+
+```{r}
+#| label: tlr
+tlr <-
+  tailor() %>%
+  adjust_probability_threshold(threshold = .7)
+
+tlr
+```
+
+tailors must be fitted before they can predict on new data. For adjustments like `adjust_probability_threshold()`, there's no training that actually happens at the `fit()` step besides recording the name and type of relevant variables. For other adjustments, like numeric calibration with `adjust_numeric_calibration()`, parameters are actually estimated at the `fit()` step and separate data should be used to train the postprocessor and evaluate its performance. More on this in [Tailors in context](#tailors-in-context).
```

Comment:
just to avoid confusion since we mentioned recipes and steps

```suggestion
tailors must be fitted before they can predict on new data. For adjustments like `adjust_probability_threshold()`, there's no training that actually happens at the `fit()` step besides recording the name and type of relevant variables. For other adjustments, like numeric calibration with `adjust_numeric_calibration()`, parameters are actually estimated at the `fit()` stage and separate data should be used to train the postprocessor and evaluate its performance. More on this in [Tailors in context](#tailors-in-context).
```